### PR TITLE
[IE-537] Make sure gem is Ruby 3 ready

### PR DESCRIPTION
**Pull request by wilkinsondesouza**
To make sure Konekti dependencies are ready for Ruby 3 we are updating their stacks and fixing any problems we find in the way. 

This PR aims to adjust gems to new pattern that we are setting with dev options and a unified CI file.

### CODE REVIEW

File: mavenlint.gemspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Gem::Specification.new do |s|
   s.name        = 'mavenlint'
-  s.version     = '1.6.0'
+  s.version     = '1.7.0'
```
Comment (wilkinsondesouza) [inline mavenlint.gemspec:5]: I'm bumping a minor version for the dependencies changes we can leave 1.6 to be a way to adjust projects before Ruby 3 and 1.7 and beyond is aimed to fix Ruby 3+ projects.

File: mavenlint.gemspec [hunk 2]
```diff
@@ -14,5 +14,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'rspec', '3.12'
   s.add_development_dependency 'rubocop', '1.31'
 
-  s.required_ruby_version = '>= 2.7.5'
+  s.required_ruby_version = '>= 3.0.0'
```
Comment (ncperry) [inline mavenlint.gemspec:17]:
```suggestion
  s.required_ruby_version = '~> 3'
```
Having updated a lot of dependencies the last few weeks, I think we should use pessimistic locking for the dependencies of our gems. Ruby likely won't have breaking changes in 3.x.x, but in a major release they might release breaking changes again. We have it green right now. In five years when someone comes back in and works on it, we'd like for them to be able to start from green.

File: .github/workflows/rspec.yml [hunk 3]
```diff
@@ -1,24 +1,59 @@
...
+
+      - name: Bundle install
+        env:
+          RAILS_ENV: test
+        run: |
+          gem install bundler
+          bundle install
+
+      - name: Run Rubocop
+        run: bundle exec rubocop
```
[the remark this answers is not in the document]
Comment (wilkinsondesouza) [reply .github/workflows/rspec.yml:59]: The linter that linted the linter first

File: mavenlint.gemspec (hunk 2 above)
Comment (wilkinsondesouza) [reply mavenlint.gemspec]: `~> 3` will do the same as `>= 3.0.0` here, if we want to be pessimistic we need to make greater than 3 and lesser than 4. WDYT?

Comment (wilkinsondesouza) [reply mavenlint.gemspec]: > ~> 1.1 means that when you bundle install, you’ll get the highest-released gem version of thin between the range >= 1.1 and < 2.0.

The article doesn't mention the case we used, but I got information from the [source](https://bundler.io/guides/gemfile.html)

> ~> 0 is identical to >= 0.0 and < 1.0.

But, nonetheless, project [rubocop ruby check does not accept this pattern](https://github.com/mavenlink/mavenlint/actions/runs/7835062981/job/21379618282), it forces to use with at least the minor.

So, I changed to use the `~> 3.0`. WDYT?

Comment (ncperry) [reply mavenlint.gemspec]: Yeah I think you're right. The `~> 3.0` should give us exactly what we want. Thanks for figuring that out.

image

**Review by ncperry — APPROVED** [review]
Looks real good

File: .devcontainer/.devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (ncperry) [inline .devcontainer/.devcontainer/devcontainer.json:1]: I think you might have nested the directory structure one too deep. I think it's supposed to just be:
`.devcontainer/devcontainer.json`

**Pull request merged**